packages/expo-image-picker/ios/ImagePickerHandler.swift

```diff
@@ -54,35 +52,25 @@ internal class ImagePickerHandler: NSObject,
   // MARK: - UIImagePickerControllerDelegate
 
   func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: MediaInfo) {
-    DispatchQueue.main.async {
-      picker.dismiss(animated: true) { [weak self] in
-        self?.handlePickedMedia(mediaInfo: info)
-      }
-    }
+    handlePickedMedia(mediaInfo: info)
+    picker.dismiss(animated: true)
   }
 
   func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
-    DispatchQueue.main.async {
-      picker.dismiss(animated: true) { [weak self] in
-        self?.handlePickingCancellation()
-      }
-    }
+    handlePickingCancellation()
+    picker.dismiss(animated: true)
   }
 
   // MARK: - PHPickerViewControllerDelegate
 
-  @available(iOS 14, *)
   func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
-    DispatchQueue.main.async {
-      picker.dismiss(animated: true) { [weak self] in
```

Comment:
Probably it doesn't make sense to wait until the picker is completely dismissed – better the handle these assets right away.